### DISCUSSION TITLE
Speech length

### DISCUSSION THREAD

**Discussion by kadattack** [neonbjb/tortoise-tts#89] [Q&A] [answered]
This is an incredible project. Well done to the creator!

I have 2 questions regarding the program. 

- I noticed that the program can't really take in long scentences. It just starts speaking mumbo-jumbo after a while. Is there a variable i can tweak to change this?
- Second question i would like to ask is if there's an easy way to save the model after sampling the voice audio files? I would like to speed up the speech generation by removing whatever overhead is used to sample the same voice each time it starts.

**Comment by neonbjb** [comment, marked as answer]
Hey, thanks!

Ack on the long sentence problem. This is mostly caused by the fact that my training set-up could not accommodate training against long speech segments, so the models just get weaker in these cases. Still, it should do a pretty good job up to 15 seconds, which is more than enough time to speak even fairly long sentences. For extra-long sentences, I recommend breaking on commas, semicolons or dashes. 

You can precompile voice latents using `get_conditioning_latents.py`: https://github.com/neonbjb/tortoise-tts#generating-conditioning-latents-from-voices. Expect to shave off about half a second by doing this.

**Reply by kadattack** [reply]
Thank you for the reply. The long sentences seem to be a problem when there is text in brackets mid sentence. I'm messing with the text brake up code atm so it includes that.

**Reply by neonbjb** [reply]
Ahhh I understand now. Yeah that's probably going to get problematic with too much prompting text. Sorry. :(

**Comment by shamoon-zafar** [comment]
Here is the detailed guide on [How to Generate Longer Speech with Tortoise TTS on Colab](https://www.aicolab.xyz/2023/12/generate-longer-speech-with-tortoise-tts.html)
